rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -3231,4 +3252,91 @@ private boolean isXmlWorkflow(InputPart fileInputPart) {
                 .contains(MediaType.APPLICATION_XML.toLowerCase());
     }
 
+    @Override
+    public RestPage<String> getTaskIds(String sessionId, long from, long to, boolean mytasks, boolean running,
+            boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedRestException, PermissionRestException {
+        return getTaskIdsByTag(sessionId, null, from, to, mytasks, running, pending, finished, offset, limit);
+    }
+    
+    @Override
+    public RestPage<String> getTaskIdsByTag(String sessionId, String taskTag, long from, long to, boolean mytasks,
+            boolean running, boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedRestException, PermissionRestException {
+        Scheduler s = checkAccess(sessionId, "tasks");
+        PageBoundariesRest boundaries = getBoundaries(offset, limit);
+        Page<TaskId> page = null;
+        try {
+            page = s.getTaskIds(taskTag, from, to, mytasks, running, pending, finished,
+                    boundaries.getOffset(), boundaries.getLimit());
+            ArrayList<String> taskNames = new ArrayList<String>(page.getList().size());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "taskNames" object should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)